fabric-component-access-api-v1/src/main/java/net/fabricmc/fabric/impl/component/access/ItemComponentContextImpl.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.component.access;
+
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.component.access.v1.ItemComponentContext;
+
+@SuppressWarnings("rawtypes")
+public final class ItemComponentContextImpl extends AbstractComponentContextImpl implements ItemComponentContext {
+	private Supplier<ItemStack> stackGetter;
+	private Consumer<ItemStack> stackSetter;
+	private ServerPlayerEntity player;
+
+	@Override
+	public Supplier<ItemStack> stackGetter() {
+		return stackGetter;
+	}
+
+	@Override
+	public Consumer<ItemStack> stackSetter() {
+		return stackSetter;
+	}
+
+	@Override
+	public ServerPlayerEntity player() {
+		return player;
+	}
+
+	@Override
+	protected World getWorldLazily() {
+		return player == null ? null : player.world;
```

Comment:
why pump the world field into the `prepare` method when we just get it from the entity?

Reply:
World is guaranteed to always be present in the context, mostly as a way to get access to various other instances but also in provider/component behavior depends on it.  The prepare variant that accepts World doesn't accept a player.  It is for items that exist in the world that have an ItemStack exposed, like ItemEntity.  If you have a player then you don't provide the world.